jobs/cloud/worker.go

```diff
@@ -0,0 +1,284 @@
+			mlog.Error("Unable to save USER_LIMIT_OVERAGE_CYCLE_END_DATE count", mlog.String("worker", worker.name), mlog.String("error", nErr.Error()))
+		}
+
+		// Store today's date (first day over limit) so we can reference it later
+		t := time.Now()
+		firstDayOverVar := &model.System{Name: model.OVER_USER_LIMIT_DATE, Value: t.Format(dateLayout)}
+		err = worker.app.Srv().Store.System().SaveOrUpdate(firstDayOverVar)
+		if err != nil {
+			mlog.Error("Unable to save USER_LIMIT_OVERAGE_CYCLE_END_DATE count", mlog.String("worker", worker.name), mlog.String("error", nErr.Error()))
+		}
+		worker.LogAndSetJobSuccess(job)
+		return
+	}
+
+	// If we've reached this point, we know at some time the installation was over the limit
+
+	subCycleEndDate, err := time.Parse(dateLayout, systemValue.Value)
+	if err != nil {
+		mlog.Error("Unable to parse USER_LIMIT_OVERAGE_CYCLE_END_DATE", mlog.String("worker", worker.name), mlog.String("error", err.Error()))
+	}
+
+	now := time.Now()
+
+	daysDifference := int64(math.Floor(now.Sub(subCycleEndDate).Hours() / 24)) // Get the difference in days between now and cycle end date
+
+	// Get admin users for emailing
+	userOptions := &model.UserGetOptions{
+		Page:     0,
+		PerPage:  100,
+		Role:     model.SYSTEM_ADMIN_ROLE_ID,
+		Inactive: false,
+	}
+	sysAdmins, err := worker.app.GetUsers(userOptions)
+	switch daysDifference {
+	case 7:
+		forgiven := worker.ForgivenessCheck(int(userDifference))
+		if forgiven {
+			worker.LogAndSetJobSuccess(job)
+			return
+		}
+
+		// TODO cc support@mattermost.com for one of the emails
+		for admin := range sysAdmins {
+			worker.app.Srv().EmailService.SendOverUserSevenDayWarningEmail(sysAdmins[admin].Email, sysAdmins[admin].Locale, *worker.app.Config().ServiceSettings.SiteURL)
+		}
+	case 14:
+		forgiven := worker.ForgivenessCheck(int(userDifference))
+		if forgiven {
+			worker.LogAndSetJobSuccess(job)
+			return
+		}
+
+		// TODO cc support@mattermost.com for one of the emails
+		for admin := range sysAdmins {
+			worker.app.Srv().EmailService.SendOverUserFourteenDayWarningEmail(sysAdmins[admin].Email, sysAdmins[admin].Locale, *worker.app.Config().ServiceSettings.SiteURL)
+		}
+	case 30:
+		forgiven := worker.ForgivenessCheck(int(userDifference))
+		if forgiven {
+			worker.LogAndSetJobSuccess(job)
+			return
+		}
+
+		// TODO cc support@mattermost.com for one of the emails
+		for admin := range sysAdmins {
+			worker.app.Srv().EmailService.SendOverUserLimitThirtyDayWarningEmail(sysAdmins[admin].Email, sysAdmins[admin].Locale, *worker.app.Config().ServiceSettings.SiteURL)
+		}
+	case 90:
+		// TODO cc support@mattermost.com for one of the emails
+		overLimitDate, _ := worker.app.Srv().Store.System().GetByName(model.OVER_USER_LIMIT_DATE)
+		for admin := range sysAdmins {
+			worker.app.Srv().EmailService.SendOverUserLimitNinetyDayWarningEmail(sysAdmins[admin].Email, sysAdmins[admin].Locale, *worker.app.Config().ServiceSettings.SiteURL, overLimitDate.Value)
+		}
+	case 91:
```

Comment:
As we discussed, this would never happen (unless they shut it down after the job ran by chance). Could we create a follow up ticket for the 91 day case?

Reply:
removed the code instead of commenting. Ticket here: https://mattermost.atlassian.net/browse/MM-30406